pages/example-3-missing-authorization-api-route/vulnerable-2.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+import { BlogPost } from '../../database/blogPosts';
+import Common from './common';
+import { User } from '../../database/users';
+
+type Props = {
+  userObject: User;
+};
+
+export default function MissingAuthentication(props: Props) {
+  const [error, setError] = useState<string>();
+  const [blogPosts, setBlogPosts] = useState<BlogPost[]>([]);
+
+  useEffect(() => {
+    async function fetchInitialData() {
+      const response = await fetch(
+        '/api/example-3-missing-authorization-api-route/vulnerable',
+      );
+
+      const data = await response.json();
+
+      if ('error' in data) {
+        setError(data.error);
+        return;
+      }
+
+      setError(undefined);
+
+      setBlogPosts([]);
+      data.blogPosts.map(blogPost => {
+        props.userObject.id == blogPost.userId &&
```

Comment:
We teach using `===` for comparisons, so we should probably also do this in the code here

Reply:
sure, np